File: JavaBasic/166_second_smallest.java
// 166	166_second_smallest.java	Find the second smallest element in an array.
public class _166_second_smallest {
    public static void main(String[] args) {
        int[] arr = {12,35,1,10,34,2};
        int min=Integer.MAX_VALUE, second=Integer.MAX_VALUE;
        for (int x:arr) { if(x<min){second=min;min=x;} else if(x<second&&x!=min) second=x; }
        System.out.println("Second smallest: " + second);
    }
}

File: JSBASIC/166_second_smallest.js
// 166	166_second_smallest.js	Find the second smallest element in an array.

let arr = [12, 35, 1, 10, 34, 1];
let sorted = [...new Set(arr)].sort((a, b) => a - b);
console.log("Array:", arr);
console.log("Second smallest:", sorted[1]);
